### ISSUE TITLE
Read J, Z, calculate Q

### ISSUE THREAD

**Issue by rlongland** [rlongland/RatesMC#11]
Following on from #9.

We should also read spins, charge, and calculate the separation energies from either AME or NuBase. Should be a quite simple extension but design is the question:

1. Needs to be backward compatible
2. All or nothing? i.e., Does the user get to mix and match whether to use AME or their own inputs?
3. Read nuclide names from title line?
4. Error checking e.g. do we input like below and check for consistency?
```
  35Ar(p,g)36K
  ****************************************************************************************************************
  1H               ! Zproj
  35Ar           ! Ztarget
  0               ! Zexitparticle (=0 when only 2 channels open)
  1H              ! Aproj   (input either mass in amu, or the nuclide's name, e.g. 23Na)
  35Ar            ! Atarget (input either mass in amu, or the nuclide's name, e.g. 23Na)
  0               ! Aexitparticle (=0 when only 2 channels open)
  ```

**Comment by rlongland** [comment]
Z, A, and J are read. Completed with commit 4181c389ddb31483f81f81bed17fd7238fb771b6

**Comment by rlongland** [comment]
Still needs Q-value calculation

**Comment by rlongland** [comment]
I have a suggestion for calculating separation energy. Enter the compound nucleus if you want it aut-calculated:
  ```
39K(p,g)40Ca
****************************************************************************************************************
1H              ! Zproj     (or nuclide's name to use AME 2020)                                                
39K             ! Ztarget                                                                                      
4He             ! Zexitparticle (=0 when only 2 channels open)                                                 
1H              ! Aproj     (input either nuclear mass in amu, or the nuclide's name, e.g. 23Na)               
39K             ! Atarget                                                                                      
4He             ! Aexitparticle (=0 when only 2 channels open)                                                 
1H              ! Jproj     (or nuclide's name to use NuBase 2020)                                             
39K             ! Jtarget                                                                                      
4He             ! Jexitparticle (=0 when only 2 channels open)                                                 
40Ca            ! projectile separation energy (keV)                                                           
7039.762        ! exit particle separation energy (=0 when only 2 channels open)                                 
  ```

**Comment by rlongland** [comment]
Fixed with commit ac0c561c8ea1ac637a13c1196d8a1bb63b8916b5
